nav2_controller/plugins/simple_progress_checker.cpp

```diff
@@ -12,33 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "nav2_controller/progress_checker.hpp"
+#include "nav2_controller/plugins/simple_progress_checker.hpp"
 #include <cmath>
+#include <string>
 #include "nav2_core/exceptions.hpp"
 #include "nav_2d_utils/conversions.hpp"
 #include "geometry_msgs/msg/pose_stamped.hpp"
 #include "geometry_msgs/msg/pose2_d.hpp"
 #include "nav2_util/node_utils.hpp"
+#include "pluginlib/class_list_macros.hpp"
 
 namespace nav2_controller
 {
 static double pose_distance(const geometry_msgs::msg::Pose2D &, const geometry_msgs::msg::Pose2D &);
 
-ProgressChecker::ProgressChecker(const rclcpp_lifecycle::LifecycleNode::SharedPtr & node)
-: nh_(node)
+void SimpleProgressChecker::initialize(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & node,
+  const std::string & plugin_name)
 {
+  nh_ = node;
   nav2_util::declare_parameter_if_not_declared(
-    nh_, "required_movement_radius", rclcpp::ParameterValue(0.5));
+    nh_, plugin_name + ".required_movement_radius", rclcpp::ParameterValue(0.5));
   nav2_util::declare_parameter_if_not_declared(
-    nh_, "movement_time_allowance", rclcpp::ParameterValue(10.0));
+    nh_, plugin_name + ".movement_time_allowance", rclcpp::ParameterValue(10.0));
   // Scale is set to 0 by default, so if it was not set otherwise, set to 0
-  nh_->get_parameter_or("required_movement_radius", radius_, 0.5);
+  nh_->get_parameter_or(plugin_name + ".required_movement_radius", radius_, 0.5);
   double time_allowance_param = 0.0;
-  nh_->get_parameter_or("movement_time_allowance", time_allowance_param, 10.0);
+  nh_->get_parameter_or(plugin_name + ".movement_time_allowance", time_allowance_param, 10.0);
   time_allowance_ = rclcpp::Duration::from_seconds(time_allowance_param);
 }
 
-void ProgressChecker::check(geometry_msgs::msg::PoseStamped & current_pose)
+void SimpleProgressChecker::check(geometry_msgs::msg::PoseStamped & current_pose)
```

Comment:
I wonder now that this is a nav2_controller plugin if `check` should be renamed to something else and return a bool for true / false rather than throwing an exception. Do you think the exception is the right logic here still now that this is at the highest level?

Reply:
Ya makes sense. Done.